### PR TITLE
Feature/#65 Add user signin email confirmation

### PR DESCRIPTION
**Pull request by MangriMen**

### CODE REVIEW

**Comment by MangriMen** [comment]
image

image

**Comment by MangriMen** [comment]
[Add user signin email confirmation](https://trello.com/c/uKDp66bL/65-add-user-signin-email-confirmation)

File: src/ducks/auth/index.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { PayloadAction, createSlice } from '@reduxjs/toolkit';
 import { LoginPayload, RegisterPayload } from './interfaces';
 
 const initialState = {
```
Comment (StepanovRoma) [inline src/ducks/auth/index.ts]: Need types for initial state

File: src/ducks/auth/sagas.ts [hunk 2]
```diff
@@ -76,11 +81,21 @@ function* logoutSaga(): Generator<
   }
 }
 
+function* sendVerificationEmailSaga(): Generator<CallEffect<void>, void, void> {
+  try {
+    yield call(sendVerificationEmail);
+    console.log('Email sent successfully');
+  } catch (err) {
+    console.error(`Error to send email: ${err}`);
```
Comment (StepanovRoma) [inline src/ducks/auth/sagas.ts]: Here we can track a positive response or an error, for this we can add 2 reducers

File: src/ducks/user/index.ts [hunk 3]
```diff
@@ -3,12 +3,13 @@ import { PayloadAction, createSlice } from '@reduxjs/toolkit';
 import { FetchUserPayload } from './interfaces';
 
 const initialState = {
```
Comment (StepanovRoma) [inline src/ducks/user/index.ts]: Need types

File: src/ducks/user/sagas.ts [hunk 4]
```diff
@@ -11,22 +11,25 @@ import {
...
 > {
   try {
+    yield call(reload);
     const response = yield call(get);
     yield put(
       fetchUserSuccess({
         username: response.displayName ?? '',
         email: response.email ?? '',
         avatar: response.photoURL ?? '',
+        isEmailVerified: response.emailVerified ?? '',
```
Comment (StepanovRoma) [inline src/ducks/user/sagas.ts]: emailVerified has boolean type, but here you write empty string, to fix this, you just need to do interfaces for initial state

**Review by StepanovRoma — CHANGES_REQUESTED** [review]

File: src/ducks/user/sagas.ts (hunk 4 above)
Comment (MangriMen) [reply src/ducks/user/sagas.ts]: Anyway it casts to boolean and interface is already there, but i change fallback value to false to more logic behavior

**Review by StepanovRoma — APPROVED** [review]

**Review by FFreestyler — APPROVED** [review]

**Pull request merged**